policies/lib/tree/30_generic_methods/file_from_template_type.cf

```diff
@@ -109,55 +109,36 @@
 bundle agent file_from_template_type(source_template, path, template_type)
 {
   vars:
-      "old_class_prefix"  string => canonify("file_from_template_${path}");
-
-      "args"               slist => { "${source_template}", "${path}", "${template_type}" };
-      "report_param"      string => join("_", args);
-      "full_class_prefix" string => canonify("file_from_template_${report_param}");
-      "class_prefix"      string => string_head("${full_class_prefix}", "1000");
-
-      # Only used in the jinja2 case
-      "full_inner_class_prefix" string => canonify("file_from_template_${source_template}_${path}");
-      "inner_class_prefix"      string => string_head("${full_inner_class_prefix}", "1000");
-
+      "class_prefix"  string => canonify("file_from_template_${path}");
       # allow to avoid resolving the jinja2 bundle when not using jinja2, because of the special requirements
       "jinja2_bundle_name" string => "file_from_template_jinja2";
 
   classes:
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
       "template_exists" expression => fileexists("${source_template}");
       "is_jinja2"       expression => strcmp("${template_type}", "jinja2");
 
-      "should_report"   expression => "${report_data.should_report}";
-
-  files:
-    "${path}"
-      create          => "true",
-      edit_template   => "${source_template}",
-      template_method => "${template_type}",
```

Comment:
this supported the cfengine templating method, we probably need to keep compatibility